File: Ruby/rankB_Up/GomokuLineUp/step5.rb
# 5行5列の五目並べの盤面が与えられます。

# 盤面の各マスには、"O"か"X"か"."が書かれています。

# "O"と"X"は、それぞれプレイヤーの記号を表します。

# 同じ記号が斜めに連続で5つ並んでいれば、その記号のプレイヤーが勝者となります。

# 勝者の記号を1行で表示してください。
# 勝者がいない場合は、引き分けとして、"D"を表示してください。

board = []
player = [ "O", "X"]
result = "D"

5.times do 
    board << gets.chomp.split("")
end

left_diagonal = []
5.times do |i|
    left_diagonal << board[i][i]
end

right_diagonal = []
5.times do |i|
    right_diagonal << board[i][4-i]
end

check_mark = []
check_mark.push(left_diagonal,right_diagonal)

player.each do |p|
    check_mark.each do |row|
        count = 0
        row.each do |i|
            if i == p
                count += 1
            end
        end
        
        if count == 5
            result = p
        end
    end
end

puts result

File: Python/ConditionalBranch/ProductMinimization/step5.py
# 2 以上の整数N, Kが与えられます。
# 「Nを 2 倍した数でNを更新する」という操作を何度か繰り返すことを考えます。
# このとき, ちょうどある回数 M で N の値は K 以上になります。この時点で操作の繰り返しを終了することにします。

# この繰り返しの回数Mを求めてください。(Mは 0 でも構いません)

# 例えば、Nを 2 、Kを 18 とします。
# 上記の操作を 3 回繰り返すと、N = 16 となります。
# 4 回くりかえすと、N = 32 となり、N >= K が成立します。ここで操作を終了し、最終的な操作の回数Mは 4 となります。

n,k = map(int,input().split())

count = 0
while n < k:
    n = n * 2
    count += 1

print(count)
